zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -812,11 +812,13 @@ public void finishSessionInit(ServerCnxn cnxn, boolean valid) {
             cnxn.sendBuffer(bb);
 
             if (valid) {
-                LOG.debug("Established session 0x"
-                        + Long.toHexString(cnxn.getSessionId())
-                        + " with negotiated timeout " + cnxn.getSessionTimeout()
-                        + " for client "
-                        + cnxn.getRemoteSocketAddress());
+                if (LOG.isDebugEnabled()) {
```

Comment:
Same here.

Reply:
The idea with parameter logging in SLF4J is that it avoids doing any real work unless the specified log-level is met.  In this example, parameters could of course be used, but the code would still call `Long.toHexString(sessionId)` for every iteration and the results of that work would be thrown away (unless debug logging is enabled).  It also has to deal with that conditional.  It's better just to skip this code altogether if debug logs are disabled.